README.md

```diff
@@ -126,375 +126,122 @@ If your car is not listed above, then there has been no documented information o
 * [Key Extraction](#key-extraction) if you don't know the key.
 * [Key Installation](#key-installation) if you know the key.
 
-> [!NOTE]
-> The key will change if you get a new bumper, because the bumper has distance sensors that use the security key. Instead of applying the existing key to the bumper, they replace the key on all parts of the car. The same goes for many other parts with secoc components.
->
-> If you never get into an accident, then the key will never change, unless a Toyota service technician presses a wrong button.
-
-
 ## Key Extraction
 
 Your car has a security key that Toyota doesn't want you to have. \
 Follow this guide to run a [hardware exploit](https://icanhack.nl/blog/secoc-key-extraction/) to extract the key.
 
-### Step 1. Upgrade AGNOS by installing `commaai/nightly-dev`
+### Step 1. Install `TSK Manager`
 
-AGNOS is the operating system used in C3X. The latest one is needed to run TSK Manager.
+At home, turn on C3X with your phone charger. Ignore the low voltage warning.
 
-1-1. At home, turn on C3X with your phone charger. Ignore the low voltage warning. USB A-to-C cables work well, and USB PD (Power Delivery) sometimes doesn't work. If all fails, you can do this in your car.
+Don't choose `Install openpilot`. Instead, choose `Custom Software` with URL `tsk.lvin.ca`
 
-1-2. Connect C3X to your Wi-Fi network.
+![](img/v3.install.jpg)
 
-1-3. Don't choose `Install openpilot`. Instead, choose `Custom Software` with URL `commaai/nightly-dev`
+![](img/v3.tsk-manager.home.jpg)
 
-The installation takes 1~5 minutes with one or two restarts.
+Unplug the power to turn off the device.
 
-If you're doing this in your car, keep the engine running to keep the 12V battery alive.
+<details><summary>Troubleshooting</summary>
 
-1-4. Scroll and accept the EULA, and go through the training.
-
-![](img/v2.nightly-dev.jpg)
+1. USB A-to-C cables work well, and USB PD (Power Delivery) sometimes doesn't work.
+2. The installation takes about 2 minutes, or longer if an OS update is needed.
+3. In some cases the installation gets stuck in "registering device" screen. If this happens, unplug the device to power off, plug it back in, and then tap-tap-tap on the screen as it boots to reset the device. Afterward, install `tsk.lvin.ca`
+</details>
 
 ### Step 2. Install the hardware
 
-2-1. Go to your car.
+Go to your car and connect everything including Comma Power (OBD2 connector + long cable)
 
-2-2. Connect the harness to your car by following the official Setup Guide: https://comma.ai/setup/comma-3x
+> [!CAUTION]
+> The car harness sends a 12V signal instead of the usual 5V. Do not plug in anything other than C3X.
```

Comment:
Is this common for people to do?

Reply:
Hmm actually not really. I'll move it to the fold.

Reply:
Moved